applications/luci-app-wifischedule/htdocs/luci-static/resources/view/wifischedule/wifischedule.js

```diff
@@ -0,0 +1,195 @@
+'require fs';
+'require uci';
+
+function timeValidator(value, desc) {
+	if (value !== null) {
+		const matches = value.match(/^(\d\d?):(\d\d?)$/);
+		if (matches) {
+			const h = parseInt(matches[1], 10);
+			const m = parseInt(matches[2], 10);
+			if (h >= 0 && h <= 23 && m >= 0 && m <= 59) {
+				return true;
+			}
+		}
+	}
+	return _('The value %s is invalid'.format(desc));
+}
+
+return view.extend({
+	load: function() {
+		return Promise.all([
+			L.resolveDefault(fs.stat('/sbin/wifi'), null),
+			L.resolveDefault(fs.stat('/usr/bin/wifi_schedule.sh'), null),
+			L.resolveDefault(fs.exec_direct('/usr/bin/wifi_schedule.sh', [ 'getmodules' ]), null),
+			L.resolveDefault(fs.stat('/usr/bin/iwinfo'), null)
+		]);
+	},
+
+
+	render: function (data) {
+		var m, s, o, oUnloadmodules, oModules;
+
+		m = new form.Map('wifi_schedule', _('Wifi Schedule'),_('Defines a schedule when to turn on and off wifi.'));
+
+		s = m.section(form.TypedSection, 'global', _('Manual control'));
+		s.optional = false;
+		s.rmempty = false;
+		s.anonymous = true;
+
+		o = s.option(form.Button, '', _('Activate wifi'));
+		o.onclick = function (section, value) {
+			fs.exec('/usr/bin/wifi_schedule.sh', ['start', 'manual']);
+		}
+
+		o = s.option(form.Button, '', _('Disable wifi gracefully'));
+		o.onclick = function (section, value) {
+			fs.exec('/usr/bin/wifi_schedule.sh', ['stop', 'manual']);
+		}
+
+		o = s.option(form.Button, '', _('Disable wifi forced'));
+		o.onclick = function (section, value) {
+			fs.exec('/usr/bin/wifi_schedule.sh', ['forcestop', 'manual']);
+		}
+
+		s = m.section(form.TypedSection, 'global', _('Global Settings'));
+		s.optional = false;
+		s.rmempty = false;
+		s.anonymous = true;
+
+		o = s.option(form.Flag, 'enabled', _('Enable Wifi Schedule'));
+		o.optional = false;
+		o.rmempty = false;
+		o.validate = function(section_id, value) {
+			if (value == '1') {
+				if (data[0] && data[1]) {
+					return true;
+				} else {
+					return _('Could not find required /usr/bin/wifi_schedule.sh or /sbin/wifi');
+				}
+			} else {
+				return true;
+			}
+		};
+
+		o = s.option(form.Flag, 'logging', _('Enable logging'));
+		o.optional = false;
+		o.rmempty = false;
+		o.default = 0;
+
+		o = s.option(form.Flag, 'unload_modules', _('Unload Modules (experimental; saves more power)'));
+		o.optional = false;
+		o.rmempty = false;
+		o.default = 0;
+
+		o = s.option(form.TextValue, 'modules', 'Modules to unload')
+		o.depends('unload_modules', '1');
+		o.wrap = 'off';
+		o.rows = 10;
+		o.cfgvalue = function (section) {
+			let mod = uci.get('wifi_schedule', section, 'modules');
+			if (mod === undefined || mod === null) {
+			  mod = "";
+			}
+			return mod.replace(/ /g, "\r\n");
+		}
+		o.write = function (section, value) {
+				var valueList = value.replace(/[\r\n]+/g, " ").replace(/\s+$/, '');
+				return uci.set('wifi_schedule', section, 'modules', valueList);
+		}
+
+		o = s.option(form.TextValue, 'detectedmodules', _('Modules detected'));
```

Comment:
would this not be better as a `DummyValue`?

Reply:
Can do.